openmdao/docs/usr-guide/basics.rst

```diff
@@ -277,22 +276,35 @@ The general procedure for defining and solving a `Problem` in OpenMDAO is:
     - perform *setup* on the `Problem` to initialize all vectors and data transfers
     - perform *run* on the Problem
 
-A very basic example of defining and running a `Problem` as discussed here is shown below.
-This example makes use of a couple of convenience components to provide a source for the
-parameter (`IndepVarComp`) and to quickly define a `Component` for an equation (`ExecComp`).
+A very basic example of defining and running a `Problem` with a custom `Component` is shown below.
+This example makes use of the convenience component `IndepVarComp` to provide a source for the
+input parameter to the custom `MultiplyByTwoComponent`.
 
 ::
 
-    from openmdao.core import Group, Problem
-    from openmdao.components import IndepVarComp, ExecComp
+    from openmdao.core import Group, Problem, Component
+    from openmdao.components import IndepVarComp
+
+    class MultiplyByTwoComponent(Component):
+        def __init__(self):
+            super(MultiplyByTwoComponent, self).__init__() # always call the base class constructor first
+            self.add_param('x_input', val=0.) # the input that will be multiplied by 2
+            self.add_output('y_output', shape=1) # shape=1 => a one dimensional array of length 1 (a scalar)
+            self.add_state('counter', val=0) # an internal variable that counts the number of times this component was executed
```

Comment:
I am a little late on this, but I am not sure we want to add an integer state. States have a special meaning in OpenMDAO as implicit variables that higher-level solvers try to converge.